python/test/unit/operators/test_matmul.py

```diff
@@ -137,22 +146,34 @@ def maybe_upcast(x, dtype, is_float8):
             return f8_to_f16(x, dtype)
         return x
 
-    def init_input(m, n, dtype):
+    string_to_torch_type = {"float16": torch.float16, "bfloat16": torch.bfloat16, "float32": torch.float32}
+
+    def init_input(m, n, dtype, only_small_integers=False):
         if 'float8' in dtype:
             ewidth = {'float8e4b15': 4, 'float8e4nv': 4, 'float8e5': 5}[dtype]
             sign = torch.randint(2, size=(m, n), device="cuda", dtype=torch.int8) * 128
             val = torch.randint(2**3 - 1, size=(m, n), device="cuda", dtype=torch.int8) << 7 - ewidth
             return sign | val
         if dtype == "int8":
+            if only_small_integers:
+                return torch.randint(-2, 2, (m, n), device="cuda", dtype=torch.int8)
             return torch.randint(-128, 127, (m, n), device="cuda", dtype=torch.int8)
-        dtype = {"float16": torch.float16, "bfloat16": torch.bfloat16, "float32": torch.float32}[dtype]
-        exponents = torch.randint(-10, 0, size=(m, n))
-        ret = (2. ** exponents).to(dtype).to("cuda")
```

Comment:
They are not integers. Exponents go from -10 to 0, so from 1/1024 to 1.

Reply:
Ah, you are right! It's the same than with integers. Thanks for pointing me this out, I could use -3 to 0 for small cases (instead of 0 to 3) :)

However, I tried to generate randomly the sign and we move from 4 failing tests to 14 failing ones :(.

The new failing tests are ones with `bfloat16`, but also smaller cases (32x32) that before only failed with output_type `float32` but now also fails with `float16`:
```FAILED test/unit/operators/test_matmul.py::test_op[32-32-32-1-1-2-None-None-None-False-False-float16-float16-True-True-float16-None] - AssertionError: Tensor-likes are not close!```

 If I force the sign to be always negative results also in 4 failing tests, so no idea what could be happening. But also I really doubt that this will help since the positive and the negative numbers shouldn't cancel each other in this random generation.

Reply:
I would propose to have float16 case with -3 to 0, what do you think?

Reply:
Offline, we decided to use positives from -4 to 0, since we shouldn't be testing precision issues in this test